src/hotspot/share/logging/logAsyncWriter.hpp

```diff
@@ -124,14 +124,13 @@ typedef KVHashtable<LogFileOutput*, uint32_t, mtLogging> AsyncLogMap;
 // instance() is MT-safe and returns the pointer of the singleton instance if and only if async logging is enabled and has well
 // initialized. Clients can use its return value to determine async logging is established or not.
 //
-// The basic operation of AsyncLogWriter is enqueue(). 2 overloading versions of it are provided to match LogOutput::write().
+// enqueue() is the basic operation of AsyncLogWriter. 2 overloading versions of it are provided to match LogOutput::write().
 // They are both MT-safe and non-blocking. Derived classes of LogOutput can invoke the corresponding enqueue() in write() and
 // return 0. AsyncLogWriter is responsible of copying neccessary data.
 //
-// The static member function flush() is designated to flush out all pending messages when JVM is terminating.
-// In normal JVM termination, flush() is invoked in LogConfiguration::finalize(). flush() is MT-safe and can be invoked arbitrary
-// times. It is no-op if async logging is not established.
-//
+// flush() is designated to flush out all pending messages. MT-safety is not provided. It is no-op if async logging is not in use.
+// In normal JVM termination, flush() is invoked in LogConfiguration::finalize(). When the users change logging configurations
+// via jcmd, LogConfiguration::configure_output() invokes it with the protection of ConfigurationLock.
```

Comment:
Suggestion:
// flush() ensures that all pending messages have been written out before it returns. It is not MT-safe
// in itself. When users change the logging configuration via jcmd, LogConfiguration::configure_output()
// calls flush() under the protection of the ConfigurationLock. In addition flush() is called during JVM
// termination, via LogConfiguration::finalize, but that occurs during the VM termination safepoint and so
// no calls to LogConfiguration::configure_output() can be active at the same time.

Reply:
If you don't want to use my wording at least changed "designated" to "designed" as "designated is the wrong word. Thanks.